From 05f753b032b310baa3eb9032d83dece10c314cc3 Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Thu, 7 Mar 2019 09:07:41 +0700
Subject: [PATCH 088/175] arm64: r8a774c0: Add thermal node and thermal zone

This patch adds Thermal node and Thermal zone for R8A774C0 SoC

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 arch/arm64/boot/dts/renesas/r8a774c0.dtsi | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a774c0.dtsi b/arch/arm64/boot/dts/renesas/r8a774c0.dtsi
index 6eef9f4..bc83804 100644
--- a/arch/arm64/boot/dts/renesas/r8a774c0.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a774c0.dtsi
@@ -780,5 +780,36 @@
 			resets = <&cpg 328>;
 			status = "disabled";
 		};
+
+		thermal: thermal@e6190000 {
+			compatible = "renesas,thermal-r8a774c0";
+			reg = <0 0xe6190000 0 0x10>, <0 0xe6190100 0 0x38>;
+			interrupts =    <GIC_SPI 67 IRQ_TYPE_LEVEL_HIGH>,
+					<GIC_SPI 68 IRQ_TYPE_LEVEL_HIGH>,
+					<GIC_SPI 69 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&cpg CPG_MOD 522>;
+			power-domains = <&sysc R8A774C0_PD_ALWAYS_ON>;
+			resets = <&cpg 522>;
+			#thermal-sensor-cells = <0>;
+		};
+
+		thermal-zones {
+			cpu-thermal {
+				polling-delay-passive = <250>;
+				polling-delay = <1000>;
+				thermal-sensors = <&thermal>;
+
+						trips {
+						cpu-crit {
+							temperature = <120000>;
+							hysteresis = <2000>;
+							type = "critical";
+						};
+					};
+
+				cooling-maps {
+				};
+			};
+		};
 	};
 };
-- 
2.7.4

